test/makeLatexReport.m

```diff
@@ -0,0 +1,209 @@
+                     stat.saveStage.error, ...
+                     stat.tikzStage.error]
+              if err
+                  fprintf(fh, ' & \\textcolor{red}{failed}');
+              else
+                  fprintf(fh, ' & \\textcolor{green!50!black}{passed}');
+              end
+          end
+      end
+      fprintf(fh, ' \\\\\n');
+  end
+  texfile_tab_completion_finish(fh);
+
+  % Write the error messages to the LaTeX file if there are any
+  if errorHasOccurred(status)
+      fprintf(fh, '\\section*{Error messages}\n\\scriptsize\n');
+      for k = 1:length(status)
+          stat = status{k};
+          testNumber = stat.index;
+          if isempty(stat.plotStage.message) && ...
+             isempty(stat.saveStage.message) && ...
+             isempty(stat.tikzStage.message)
+              continue % No error messages for this test case
+          end
+
+          fprintf(fh, '\n\\subsection*{Test case %d: \\texttt{%s}}\n', testNumber, name2tex(stat.function));
+          print_verbatim_information(fh, 'Plot generation', stat.plotStage.message);
+          print_verbatim_information(fh, 'PDF generation' , stat.saveStage.message);
+          print_verbatim_information(fh, 'matlab2tikz'    , stat.tikzStage.message);
+      end
+      fprintf(fh, '\n\\normalsize\n\n');
+  end
+
+  % now, finish off the file and close file and window
+  texfile_finish(fh, status);
+  fclose(fh);
+end
+% =========================================================================
+function texfile_init(texfile_handle)
+
+  fprintf(texfile_handle, ...
+           ['\\documentclass[landscape]{scrartcl}\n'                , ...
+            '\\pdfminorversion=6\n\n'                               , ...
+            '\\usepackage{amsmath} %% required for $\\text{xyz}$\n\n', ...
+            '\\usepackage{hyperref}\n'                              , ...
+            '\\usepackage{graphicx}\n'                              , ...
+            '\\usepackage{epstopdf}\n'                              , ...
+            '\\usepackage{tikz}\n'                                  , ...
+            '\\usetikzlibrary{plotmarks}\n\n'                       , ...
+            '\\usepackage{pgfplots}\n'                              , ...
+            '\\pgfplotsset{compat=newest}\n\n'                      , ...
+            '\\usepackage[margin=0.5in]{geometry}\n'                , ...
+            '\\newlength\\figurewidth\n'                            , ...
+            '\\setlength\\figurewidth{0.4\\textwidth}\n\n'          , ...
+            '\\begin{document}\n\n']);
+
+end
+% =========================================================================
+function texfile_finish(texfile_handle, status)
+
+  [env,versionString] = getEnvironment();
+  
+  testsuites = unique(cellfun(@(s) func2str(s.testsuite) , status, ...
+                     'UniformOutput', false));                 
+  testsuites = name2tex(strjoin(testsuites, ', '));
```

Comment:
[`strcat`](http://nl.mathworks.com/help/matlab/ref/strcat.html?refresh=true) is string concatenation. `strjoin` is string joining, so both shouldn't be equivalent.

But I agree on your comment: `strjoin` should not be used since there are reasonable alternatives (in this case: implement it ourselves, I have done it before in M2T).